homeassistant/components/device_tracker/config_entry.py

```diff
@@ -0,0 +1,126 @@
+"""Code to set up a device tracker platform using a config entry."""
+from typing import Optional
+
+from homeassistant.helpers.typing import GPSType
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.entity_component import EntityComponent
+from homeassistant.const import (
+    STATE_NOT_HOME,
+    STATE_HOME,
+    ATTR_GPS_ACCURACY,
+    ATTR_LATITUDE,
+    ATTR_LONGITUDE,
+    ATTR_BATTERY_LEVEL,
+)
+from homeassistant.components import zone
+
+from .const import (
+    ATTR_SOURCE_TYPE,
+    DOMAIN,
+    LOGGER,
+    SOURCE_TYPE_GPS,
+)
+
+
+async def async_setup_entry(hass, entry):
+    """Set up an entry."""
+    component = hass.data.get(DOMAIN)  # type: Optional[EntityComponent]
+
+    if component is None:
+        component = hass.data[DOMAIN] = EntityComponent(
+            LOGGER, DOMAIN, hass
+        )
+
+    return await component.async_setup_entry(entry)
+
+
+async def async_unload_entry(hass, entry):
+    """Unload an entry."""
+    return await hass.data[DOMAIN].async_unload_entry(entry)
+
+
+class DeviceTrackerEntity(Entity):
+    """Represent a tracked device."""
+
+    @property
+    def battery_level(self):
+        """Return the battery level of the device.
+
+        Percentage from 0-100.
+        """
+        return None
+
+    @property
+    def location_accuracy(self):
+        """Return the location accuracy of the device.
+
+        Value in meters.
+        """
+        return 0
+
+    @property
+    def location_name(self) -> str:
+        """Return a location name for the current location of the device."""
+        return None
+
+    @property
+    def location(self) -> GPSType:
+        """Return location value of the device."""
+        # For MVP, we will only support location based devices.
+        return NotImplementedError
+
+    @property
+    def mac_address(self):
```

Comment:
This is only useful for scanner based entities. At least that's why I added it.

Reply:
I'll drop it.